modules/services/window-managers/hyprland.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.wayland.windowManager.hyprland;
+  defaultHyprlandPackage = pkgs.hyprland.override {
+    enableXWayland = cfg.xwayland.enable;
+    hidpiXWayland = cfg.xwayland.hidpi;
+    nvidiaPatches = cfg.nvidiaPatches;
+  };
+in {
+  meta.maintainers = [ lib.maintainers.fufexan ];
+
+  options.wayland.windowManager.hyprland = {
+    enable = lib.mkEnableOption (lib.mdDoc "Hyprland wayland compositor");
+
+    package = lib.mkOption {
+      type = with lib.types; nullOr package;
+      default = defaultHyprlandPackage;
+      description = lib.mdDoc ''
+        Hyprland package to use. Will override the 'xwayland' and 'nvidiaPatches'
+        options.
```

Comment:
Seems a bit confusing, why not have the default be `pkgs.hyprland` and then add a read-only `finalPackage` option (like in the emacs module, for example).

In the config section set

    wayland.windowManager.hyprland.finalPackage = cfg.package.override { … }

Reply:
Oh, I didn't even think of this. Thanks, will fix.

Reply:
Updated.